runbooks/how-to-update-for-rda-api-changes.md

```diff
@@ -0,0 +1,53 @@
+# How to Update for RDA API Changes
+
+1. Synch RDA API changes
+   1. Clone/pull latest changes from RDA API repo
+       - https://github.cms.gov/medicare-payment-modernization/rda-api
+   2. Copy changes from RDA API proto files to BFD proto files
+        - RDA API: https://github.cms.gov/medicare-payment-modernization/rda-api/tree/main/proto/src/main/proto/rda/v1
+        - BFD: https://github.cms.gov/medicare-payment-modernization/rda-api/tree/main/proto/src/main/proto/rda/v1
+   3. Update VERSION.txt in BFD repo
+        - https://github.com/CMSgov/beneficiary-fhir-data/tree/master/apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/proto/rda/VERSION.txt
+2. Update RDA schema/entity mappings
+   1. Diff changes between RDA and BFD proto files
+   2. Make changes to appropriate YAML mapping files
+        - https://github.com/CMSgov/beneficiary-fhir-data/tree/master/apps/bfd-model/bfd-model-rda/mappings
+        - **ALL** properties/classes should be captured, even if ultimately not served to end users.
+        - New properties get added to existing YAML mapping files
+        - New proto files get added as new YAML mapping files
```

Comment:
Maybe add these conventions:

- Column names in tables generally match corresponding field names in proto file.
- Field names in entities generally match corresponding field names in generated gRPC stub objects.
- Mapping `id`s in yaml file generally match message name in proto file.
- Entity class names generally match message name with `Rda` prefix.
- RDA generally encodes dates as strings in `YYYY-MM-DD` format but we transform to `LocalDate`.
- Fields marked as `optional` in proto file are mapped to `nullable` columns and are stored as `null` in database when not present in a given API message.
- RDA generally does not send `optional` fields that would otherwise have empty string values.  We store those as `null` in database.
- RDA generally wraps enum fields in `oneof` (basically a union) along with a field to contain text value if it did not match an enum value.  All `oneof` fields can actually have no value (i.e. all components are basically optional and protobuf does not require at least one to have a value).
- Look for similar fields in existing mappings to see how to handle enums and other non-string types.